include/eve/module/core/regular/swap_adjacent_groups.hpp

```diff
@@ -7,12 +7,96 @@
 //==================================================================================================
 #pragma once
 
-#include <eve/arch.hpp>
-#include <eve/detail/overload.hpp>
-#include <eve/pattern.hpp>
+#include <eve/detail/shuffle_v2/named_shuffle.hpp>
+#include <eve/module/core/regular/shuffle_v2.hpp>
 
 namespace eve
 {
+
+//================================================================================================
+//! @addtogroup core
+//! @{
+//!   @var swap_adjacent
+//!   @brief goes through all pairs of elements and them: [0, 1, 2, 3] => [1,0,3, 2]
+//!   Can do that for group sizes instead of elements.
+//!   If group size is equal to the register size - that's a noop.
+//!
+//!   @code
+//!   template<simd_value T, std::ptrdiff_t G>
+//!   T swap_adjacent(T x, fixed<G> const &) noexcept;  // (1)
+//!
+//!   template<simd_value T>
+//!   T swap_adjacent(T x) noexcept;  // (2)
+//!   @endcode
+//!
+//!   (2) calls (1) with G == 1;
+//!   (1) swaps adjacent groups of elements of size G
+//!
+//!  **Return value**
+//!
+//!    Return x where groups with contiguous groups of N elements swapped
+//!    for N =  0 or cardinal_v<T> acts as noop.
+//!  @groupheader{Example}
+//!
+//!  @godbolt{doc/core/regular/swap_adjacent_groups.cpp}
+//! @}
+//================================================================================================
+struct swap_adjacent_t
+{
+  template<simd_value T, std::ptrdiff_t G>
+  static constexpr auto pattern(eve::as<T>, eve::fixed<G>)
+  requires(G < T::size())
+  {
+    return eve::fix_pattern<T::size() / G>(
+        [](int i, int)
+        {
+          if( i & 1 ) return i - 1;
+          else return i + 1;
+        });
+  }
+
+  template<arithmetic_simd_value T, std::ptrdiff_t G>
+  static constexpr std::ptrdiff_t level(eve::as<T>, eve::fixed<G> g)
```

Comment:
I am not sure about puttiing expected level here. On one hand, it can just be in tests. On the other - the user miht want to look at it.

Reply:
sounds good to me